### ISSUE TITLE
Interpreting access lint vs Google reporting 

### ISSUE THREAD

**Issue by arthur24b6** [accesslint/accesslint.rb#17]
Hi-

I'm wondering about the differences in reporting that I'm seeing between [Google's auditing tool](https://github.com/GoogleChrome/accessibility-developer-tools) and what I'm seeing from access lint. I anticipated that they would produce the same results but I'm seeing that access lint is reporting failures that Google's audit isn't. For example:

```
$ access_lint audit http://google.com
auditing http://google.com

PASS - This element does not support ARIA roles, states and properties (Warning)

PASS - aria-owns should not be used if ownership is implicit in the DOM (Warning)

PASS - ARIA state and property values must be valid (Severe)

PASS - Controls and media elements should have labels (Severe)

PASS - Any ID referred to via an IDREF must be unique in the DOM (Severe)

PASS - The web page should have the content's human language indicated in the markup (Warning)

PASS - An element's ID must not be present in more that one aria-owns attribute at any time (Warning)

PASS - ARIA attributes which refer to other elements by ID should refer to elements which exist in the DOM (Severe)

PASS - The web page should have a title that describes topic or purpose (Warning)

PASS - This element has an unsupported ARIA attribute (Severe)

NA - Elements with ARIA roles must be in the correct scope (Severe)

NA - Audio elements should have controls (Warning)

NA - Elements with ARIA roles must use a valid, non-abstract ARIA role (Severe)

NA - role=main should only appear on significant elements (Warning)

NA - A label element may not have labelable descendants other than its labeled control. (Severe)

NA - Elements with ARIA roles must have all required attributes for that role (Severe)

NA - Elements with ARIA roles must ensure required owned elements are present (Severe)

NA - Elements with role=tooltip should have a corresponding element with aria-describedby (Severe)

NA - Avoid positive integer values for tabIndex (Warning)

NA - A tabpanel should be related to a tab via aria-controls or aria-labelledby (Warning)

NA - Video elements should use <track> elements to provide captions (Warning)

FAIL - This element has an invalid ARIA attribute (Warning)
* <div class="gbm" id="gbd" aria-owner="gbztm"><div id="gbmmb" class="gbmc gbsb gbs...
* <div class="gbm" id="gbd5" aria-owner="gbg5"><div class="gbmc"><ol id="gbom" clas...

FAIL - These elements are focusable but either invisible or obscured by another element (Warning)
* <a onclick="gbar.logger.il(1,{t:5});" class="gbzt" id="gb_5" href="http://news.go...
* <a onclick="gbar.logger.il(1,{t:23});" class="gbzt" id="gb_23" href="https://mail...

FAIL - Images should have a text alternative or presentational role (Warning)
* <img src="/images/icons/product/chrome-48.png">...

FAIL - The purpose of each link should be clear from the link text (Warning)
* <a class="gbgt" id="gbztm" href="http://www.google.com/intl/en/options/" onclick=...

FAIL - Text elements should have a reasonable contrast ratio (Warning)
* <span class="gbts">Google+</span>...
* <span class="gbts">Search</span>...
* <span class="gbts">Images</span>...
* <span class="gbts">Maps</span>...
* <span class="gbts">Play</span>...
* <span class="gbts">YouTube</span>...
* <span class="gbts">News</span>...
* <span class="gbts">Gmail</span>...
* <span id="gbztms1">More</span>...
* <span id="gbi4s1">Sign in</span>...
* <div class="xbtn" onclick="google.promos&amp;&amp;google.promos.toast&amp;&amp; g...
* <a href="/chrome/browser/?hl=en&amp;brand=CHNG&amp;utm_source=en-hpp&amp;utm_medi...

FAIL - Meaningful images should not be used in element backgrounds (Warning)
* <span class="gbts"></span>...
* <span id="gbi5"></span>...
* <input class="lsb" value="Google Search" name="btnG" type="submit">...
* <input class="lsb" value="I'm Feeling Lucky" name="btnI" onclick="if(this.form.q....

FAIL - Tables should have appropriate headers (Severe)
* <table border="0"> <tbody><tr> <td colspan="2">  </td> </tr> <tr> <td class="padi...
* <table cellpadding="0" cellspacing="0"><tbody><tr valign="top"><td width="25%">&n...
* <table cellspacing="0" cellpadding="0" style="display: none; width: 500px; top: 2...
complete
```

As compared to Google's audit: 

```
$ phantomjs tools/runner/audit.js http://google.com
*** Begin accessibility audit results ***
An accessibility audit found
Errors:
Error: AX_TABLE_01 (Tables should have appropriate headers) failed on the following elements (1 - 3 of 3):
#pmocntr2 > TABLE
CENTER > FORM > TABLE
body > .gstl_0.gssb_c
See https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_table_01 for more information.


Warnings:
Warning: AX_ARIA_11 (This element has an invalid ARIA attribute) failed on the following elements (1 - 2 of 2):
#gbd
#gbd5
See https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_aria_11 for more information.

Warning: AX_FOCUS_01 (These elements are focusable but either invisible or obscured by another element) failed on the following elements (1 - 2 of 2):
#gb_5
#gb_23
See https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_focus_01 for more information.

Warning: AX_TEXT_02 (Images should have a text alternative or presentational role) failed on the following element:
#pmocntr2 > TABLE > TBODY > TR:nth-of-type(2) > .padi > IMG
See https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_text_02 for more information.

Warning: AX_TEXT_04 (The purpose of each link should be clear from the link text) failed on the following element:
#gbztm
See https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_text_04 for more information.

Warning: AX_COLOR_01 (Text elements should have a reasonable contrast ratio) failed on the following elements (1 - 5 of 12):
#gb_119 > .gbts
#gb_1 > .gbts
#gb_2 > .gbts
#gb_8 > .gbts
#gb_78 > .gbts
See https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_color_01 for more information.

Warning: AX_IMAGE_01 (Meaningful images should not be used in element backgrounds) failed on the following elements (1 - 4 of 4):
#gbg > .gbtc > .gbt.gbtb > .gbts
#gbi5
CENTER > FORM > TABLE > TBODY > TR > TD:nth-of-type(2) > SPAN > .lsbb > .lsb
CENTER > FORM > TABLE > TBODY > TR > TD:nth-of-type(2) > SPAN:nth-of-type(2) > .lsbb > .lsb
See https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_image_01 for more information.


*** End accessibility audit results ***
```

Is it the case that I'm simply wrong to expect similar results? Thanks again for your time!

**Comment by ckundo** [comment]
Hi @arthur24b6, the audit rules in access_lint are from HEAD on master of the Google tools. It's possible that you're running an outdated version of Google's tool?

**Comment by arthur24b6** [comment]
I checked out the master branch of the Google tools and diffed that against the axs_testing.js which showed them to be the same- I'll poke at it again tomorrow to ensure that that was the case.
